DEPRECATION.md

```diff
@@ -85,75 +78,130 @@ here][link-enable-https].
+microservices, if necessary).
 
-[link-nodejs-eol]: https://github.com/nodejs/Release#end-of-life-releases
+[link-cube-docker]: https://cube.dev/blog/cubejs-loves-docker
+[link-migration]:
+  https://cube.dev/docs/configuration/overview#migrating-from-express-to-docker
 
-### Absolute import for @cubejs-backend/query-orchestrator
+### Absolute import for `@cubejs-backend/query-orchestrator`
 
 **Deprecated in Release: v0.24.2**
 
-Reason: Absolute imports highly depend on a path, and all API becomes public. Now we started to provide public API as `export` from the module.
+Absolute imports highly depend on a path, and all API becomes public. We now
+provide a public API from the package directly.
 
 Deprecated:
 
-```js
+```javascript
 const BaseDriver = require('@cubejs-backend/query-orchestrator/driver/BaseDriver');
 ```
 
 You should use:
 
-```js
+```javascript
 const { BaseDriver } = require('@cubejs-backend/query-orchestrator');
 ```
 
-### Absolute import for @cubejs-backend/server-core
+### `contextToDataSourceId`
+
+**Removed in Release: v0.25.0**
+
+The `contextToDataSourceId` option in the `cube.js` configuration file has been
+replaced by [`contextToOrchestratorId`][link-contexttoorchestratorid]. Prior to
+this change, multi-tenant setups were forced to share a Query Orchestrator
+instance. Now orchestrator instances can be shared by Cube.js instances and
+across different tenants, if need be. Single-tenant setups should consider
+removing the `contextToDataSourceId` property completely.
+
+[link-contexttoorchestratorid]:
+  https://cube.dev/docs/config#options-reference-context-to-orchestrator-id
+
+### Absolute import for `@cubejs-backend/server-core`
 
 **Deprecated in Release: v0.25.4**
 
-Reason: Absolute imports highly depend on a path, and all API becomes public. Now we started to provide public API as `export` from the module.
+Absolute imports highly depend on a path, and all API becomes public. We now
+provide a public API from the package directly.
 
 Deprecated:
 
-```js
+```javascript
 const CubejsServerCore = require('@cubejs-backend/server-core');
 ```
 
 You should use:
 
-```js
+```javascript
 const { CubejsServerCore } = require('@cubejs-backend/server-core');
 ```
 
-### Absolute import for @cubejs-backend/schema-compiler
+### Absolute import for `@cubejs-backend/schema-compiler`
 
 **Deprecated in Release: v0.25.21**
 
-Reason: Absolute imports highly depend on a path, and all API becomes public. Now we started to provide public API as `export` from the module.
+Absolute imports highly depend on a path, and all API becomes public. We now
+provide a public API from the package directly.
 
 Deprecated:
 
-```js
+```javascript
 const BaseQuery = require('@cubejs-backend/schema-compiler/adapter/BaseQuery');
 ```
 
 You should use:
 
-```js
+```javascript
 const { BaseQuery } = require('@cubejs-backend/schema-compiler');
 ```
+
+### `checkAuthMiddleware`
```

Comment:
Already exists, https://github.com/cube-js/cube.js/pull/1799/files#diff-6fc6a5a49016824787dbfe28ea9b9b0981eefcb8118c54a6eea809bfa635ec68R164

Can we sync somehow?